lib/spack/spack/cmd/info.py

```diff
@@ -56,108 +123,97 @@ def setup_parser(subparser: argparse.ArgumentParser) -> None:
 
-        # Compute the format
-        self.fmt = "%%-%ss%%-%ss%%-%ss%%s" % (
-            self.column_widths[0] + 4,
-            self.column_widths[1] + 4,
-            self.column_widths[2] + 4,
-        )
+    Returns a list of the most used when conditions for boolean variants along with their value.
+    """
+    top: Dict = collections.defaultdict(int)
+    for when, _ in when_indexed_dictionary.items():
+        for v, variant in when.variants.items():
+            if type(variant.value) is bool:
+                top[(variant.name, variant.value)] += 1
 
-    def default(self, v):
-        s = "on" if v.default is True else "off"
-        if not isinstance(v.default, bool):
-            s = v.default
-        return s
+    # sorted by frequency, highest first
+    return list(reversed(sorted((n, t) for t, n in top.items())))
 
-    @property
-    def lines(self):
-        if not self.variants:
-            yield "    None"
-            return
 
-        else:
-            yield "    " + self.fmt % self.headers
-            underline = tuple([w * "=" for w in self.column_widths])
-            yield "    " + self.fmt % underline
-            yield ""
-            for k, e in sorted(self.variants.items()):
-                v, w = e
-                name = textwrap.wrap(
-                    "{0} [{1}]".format(k, self.default(v)), width=self.column_widths[0]
-                )
-                if all(spec == spack.spec.Spec() for spec in w):
-                    w = "--"
-                when = textwrap.wrap(str(w), width=self.column_widths[1])
-                allowed = v.allowed_values.replace("True, False", "on, off")
-                allowed = textwrap.wrap(allowed, width=self.column_widths[2])
-                description = []
-                for d_line in v.description.split("\n"):
-                    description += textwrap.wrap(d_line, width=self.column_widths[3])
-                for t in zip_longest(name, when, allowed, description, fillvalue=""):
-                    yield "    " + self.fmt % t
-
-
-def print_dependencies(pkg, args):
+def print_dependencies(pkg: PackageBase, args: Namespace) -> None:
     """output build, link, and run package dependencies"""
+    print_definitions(pkg, "Dependencies", pkg.dependencies, DependencyFormatter(), args.by_name)
 
-    for deptype in ("build", "link", "run"):
-        color.cprint("")
-        color.cprint(section_title("%s Dependencies:" % deptype.capitalize()))
-        deps = sorted(pkg.dependencies_of_type(dt.flag_from_string(deptype)))
-        if deps:
-            colify(deps, indent=4)
-        else:
-            color.cprint("    None")
+
+def print_dependency_suggestion(pkg: PackageBase) -> None:
+    variant_counts = count_bool_variant_conditions(pkg.dependencies)
+    big_variants = [
+        (name, val)
+        for n, (name, val) in variant_counts
+        # counts >= 40 and skip if the user already disabled the variant
```

Comment:
(nitpick) Let's use something generic in comments, so they don't go out of sync
```suggestion
        # counts >= <threshold> and skip if the user already disabled the variant
```